aiida/orm/nodes/data/array/kpoints.py

```diff
@@ -228,16 +219,17 @@ def set_cell(self, cell, pbc=None):
         if pbc is None:
             pbc = [True, True, True]
         self.pbc = pbc
-        self._set_reciprocal_cell()
 
-    def _set_reciprocal_cell(self):
+    @property
+    def reciprocal_cell(self):
         """
-        Sets the reciprocal cell in units of 1/Angstrom from the internally set cell
+        Return the reciprocal cell in units of 1/Angstrom from the internally set cell
         """
         the_cell = numpy.array(self.cell)
-        self.reciprocal_cell = 2. * numpy.pi * numpy.linalg.inv(the_cell).transpose()
+        reciprocal_cell = 2. * numpy.pi * numpy.linalg.inv(the_cell).transpose()
```

Comment:
Would it maybe make sense to construct this lazily and cache it? Or is it too cheap a computation to even bother

Reply:
I thought about it, I decided it's very cheap, and also "dangerous" to cache because in principle one could "re-set" the cell and then one should invalidate the cache. Possible, but probably not worth it unless we have a real usecase of the benefits.